src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -698,6 +721,12 @@ AdaptiveExecutor(CitusScanState *scanState)
 
 	FinishDistributedExecution(execution);
 
+	if (hasDependedJobsInAllTree && SubPlanLevel == 0)
+	{
+		CleanUpSchemas();
```

Comment:
This is only meant to happen at system start-up since it removes *all* job schemas.

Reply:
what is the problem with removing all job schemas? Is that because another repartition query could be in progress while this one is removing all job schemas?

Reply:
yes